src/libraries/System.Private.CoreLib/src/System/Math.cs

```diff
@@ -113,6 +114,52 @@ public static long BigMul(int a, int b)
             return ((long)a) * b;
         }
 
+        private static ulong BigMulSoftwareFallback(ulong a, ulong b, out ulong low)
+        {
+            const ulong lowBitsMask = 0xFFFFFFFFU;
+
+            ulong al = a & lowBitsMask;
+            ulong ah = a >> 32;
+            ulong bl = b & lowBitsMask;
+            ulong bh = b >> 32;
+
+            ulong mull = al * bl;
+            ulong t = ah * bl + (mull >> 32);
+            ulong tl = t & lowBitsMask;
+
+            tl += al * bh;
+            low = tl << 32 | mull & lowBitsMask;
+
+            return ah * bh + (t >> 32) + (tl >> 32);
+        }
+
+        [CLSCompliant(false)]
```

Comment:
Should this be marked as AggressiveInlining ?

Reply:
Once https://github.com/dotnet/runtime/issues/11782 is fixed